From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Stefatorus <stefanluciandeleanu@gmail.com>
Date: Sun, 11 Apr 2021 19:35:24 +0300
Subject: [PATCH] Disable light updates when tps under 18


diff --git a/src/main/java/net/minecraft/server/Chunk.java b/src/main/java/net/minecraft/server/Chunk.java
index f9a6a93fb134f05f786f162eb278b21d23cdf93a..f8951e9b1e7bcd043c46b8f9868654cba8d63c35 100644
--- a/src/main/java/net/minecraft/server/Chunk.java
+++ b/src/main/java/net/minecraft/server/Chunk.java
@@ -1321,7 +1321,14 @@ public class Chunk {
     }
 
     public void n() {
+    	
+    	
     	// 2p2t Start - 4048
+    	if (MinecraftServer.getServer().recentTps[0] < 18) {
+    		return;
+    	}
+    	// 2p2t Stop
+    	
         if (this.x < 1024) {
             BlockPosition blockposition = new BlockPosition(this.locX << 4, 0, this.locZ << 4);
 
